src/dataflow-types/src/client.rs

```diff
@@ -379,39 +419,71 @@ impl<T: timely::progress::Timestamp> ComputeCommandHistory<T> {
         // First determine what the final compacted frontiers will be for each collection.
         // These will determine for each collection whether the command that creates it is required,
         // and if required what `as_of` frontier should be used for its updated command.
+        //
+        //
+        // Note that when we reduce, we end up tracing the work under the `OpenTelemetryContext`
+        // of the final command
+        // TODO(guswynn): consider holding a durable log of just `ComputeCommandType`'s
+        // and create new otel contexts
```

Comment:
This part doesn't seem right to me? Why are we hanging on to only the last context? ISTM that we want to be shoving the attendant context into each map alongside the command.

Reply:
I think it's not clear as it isn't spec'd out yet what this member means.

Reply:
@benesch for peeks here you are right, we should track them all, but for the cancels, createdataflows, and allowcompactions, multiple commands become 1, so we have to choose one. It may very well be more clear to just create a new root span to attach each time we call actually use the history in `hydrate_replica`?